web/web.go

```diff
@@ -582,25 +583,60 @@ func (h *Handler) rules(w http.ResponseWriter, r *http.Request) {
 	h.executeTemplate(w, "rules.html", h.ruleManager)
 }
 
-func (h *Handler) targets(w http.ResponseWriter, r *http.Request) {
-	// Bucket targets by job label
+// Bucket targets by job label
+func bucketTargets(targets []*retrieval.Target) struct {
+	TargetPools map[string][]*retrieval.Target
```

Comment:
This is weird for a return type, especially because we could have returned: `map[string][]*retrieval.Target`

I see that this is because of the expectation of the targets.html template. I would suggest that you do the following below:

```
pools := bucketTargets()
executeTemplate(w, "targets.html", struct{
  TargetPools map[string][]*retrieval.Target
}{pools},
)
```

But the better thing would be to change the template to directly accept the `map[string][]*retrieval.Target`